### PR TITLE
Ignore security test that fails in CI

### PR DESCRIPTION
**Pull request by timja**
See [JENKINS-XXXXX](https://issues.jenkins-ci.org/browse/JENKINS-XXXXX).

### Proposed changelog entries

* Entry 1: Issue, Human-readable Text
* ...

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: core/src/test/java/jenkins/util/VirtualFileTest.java [hunk 1]
```diff
@@ -1000,6 +1001,7 @@ public void testCanRead_True_FileVF() throws Exception {
     }
 
     @Test
+    @Ignore("TODO doesn't pass on ci.jenkins.io due to root user being used in container tests")
```
Comment (timja) [inline core/src/test/java/jenkins/util/VirtualFileTest.java:1004]: feel free to amend this with assume or whatever, all I have time for now

**Review by daniel-beck — CHANGES_REQUESTED** [review]
Incomplete, needs to apply the same fix to `#testCanRead_False_FilePathVF`.

**Review by basil — APPROVED** [review]
Thanks for fixing this, @timja. These failing tests have prevented incrementals from being published in PRs, which has been blocking my progress on the Commons FileUpload unforking for a few days now.

**Comment by MarkEWaite** [comment]
See https://github.com/MarkEWaite/jenkins/commits/fix-ci in commit https://github.com/MarkEWaite/jenkins/commit/7f1ac320055fa7731772f9bf689e5880ec3c1c85 for an alternative implementation that replaces the `@Ignore` with a conditional in the test for a process running with permission to write to the root directory of the Unix file system

**Review by MarkEWaite — COMMENTED** [review]
See https://github.com/MarkEWaite/jenkins/commits/fix-ci in commit https://github.com/MarkEWaite/jenkins/commit/7f1ac320055fa7731772f9bf689e5880ec3c1c85 for an alternative implementation that replaces the `@Ignore` with a conditional in the test for a process running with permission to write to the root directory of the Unix file system

**Comment by timja** [comment]
> See [MarkEWaite/jenkins@`fix-ci` (commits)](https://github.com/MarkEWaite/jenkins/commits/fix-ci) in commit [MarkEWaite@7f1ac32](https://github.com/MarkEWaite/jenkins/commit/7f1ac320055fa7731772f9bf689e5880ec3c1c85) for an alternative implementation that replaces the `@Ignore` with a conditional in the test for a process running with permission to write to the root directory of the Unix file system

This fails locally for me on a Mac, changing it to:
`return System.getProperty("user.name").equals("root");` does work on my machine.

Given that CI is currently blocked, I'm going to ship the tests as ignored, if anyone wants to file a reliable follow up to un-ignore then go for it.

File: core/src/test/java/jenkins/util/VirtualFileTest.java (hunk 1 above)
Comment (olblak) [reply core/src/test/java/jenkins/util/VirtualFileTest.java:1004]: I opened [INFRA-2879](https://issues.jenkins.io/browse/INFRA-2879) to keep track of what needs to be done here. In short, we build jnlp-inbound image for every major tool, like maven,python,node,etc... and for each time we never configure the user jenkins:jenkins.

**Pull request merged**